daemon/control.c

```diff
@@ -1757,6 +1761,28 @@ control_local_new(const char *path)
 	return control;
 }
 
+#ifdef SYSTEMD
+control_t *
+control_local_new()
+{
+	control_t *control = NULL;
+	int fds = sd_listen_fds(0);
+	if (fds < 0) {
+		WARN("sd_listen_fds failed with %d", fds);
+	} else if (fds > 1) {
+		WARN("Systemd passed to many file descriptors");
+	} else if (fds == 1) {
+		control_t *control;
+		int sock = SD_LISTEN_FDS_START;
```

Comment:
i would prefer a helper function for this in common/sock.c e.g., sock_sd_listen_fd()
which returns the sock.